library/common/event/provisional_dispatcher.cc

```diff
@@ -7,29 +7,45 @@
 namespace Envoy {
 namespace Event {
 
-void ProvisionalDispatcher::drain() {
+void ProvisionalDispatcher::drain(Event::Dispatcher& event_dispatcher) {
+  // TODO(goaway): Must be called from the real dispatcher's thread, but we can't assert here because the main dispatch loop has not yet started: event_dispatcher_->isThreadSafe() will crash.
```

Comment:
Now I am a little confused. Is the original reason why we delay getting the Event::Dispatcher ref from the server because the object itself is not valid until the post init callback, not because the event loop is not running?

Reply:
Just what I observed - `post` works here, but `isThreadSafe` consistently crashed and the logs indicated the event loop didn't start running until after. That said, provided we fix stats, I think we probably don't need the postinit callback anymore.